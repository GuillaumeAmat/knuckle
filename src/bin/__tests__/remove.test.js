const binScriptPath = require('../../helpers/tests/binScriptPath');
const cmd = require('../../helpers/tests/cmd');
const getKnuckleConfig = require('../../helpers/tests/getKnuckleConfig');
const setupPristineTestFolder = require('../../helpers/tests/setupPristineTestFolder');

setupPristineTestFolder();

describe('REMOVE', () => {
  it('should fail if there are no tools in the input', async () => {
    try {
      await cmd.run('node', [binScriptPath, 'remove']);
    } catch (err) {
      expect(err).toMatchSnapshot();
    }

    expect.hasAssertions();
  });

  it('should not fail if no tools are configured', async () => {
    let hasThrown = false;

    try {
      await cmd.run('node', [binScriptPath, 'remove', 'eslint']);
    } catch (err) {
      hasThrown = true;
    }

    expect(hasThrown).toBe(false);
  });

  it('should remove tools', async () => {
    await cmd.run('node', [binScriptPath, 'add', 'prettier', 'eslint', 'lint-staged', 'husky']);
    let config = getKnuckleConfig().config;
    expect(config.tools.length).toBe(4);

    await cmd.run('node', [binScriptPath, 'remove', 'eslint']);
    config = getKnuckleConfig().config;
    expect(config.tools.length).toBe(3);
  });
});
