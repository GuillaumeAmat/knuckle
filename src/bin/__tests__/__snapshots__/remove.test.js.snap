// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REMOVE should fail if there are no tools in the input 1`] = `
"error: missing required argument 'toolname'
"
`;
